src/Dom/Document.php

```diff
@@ -818,6 +828,32 @@ public function getRemainingCustomCssSpace()
         );
     }
 
+    /**
+     * Get the array of allowed keys of lazily-created, cached properties.
+     * The array index is the key and the array value is the key's default value.
+     *
+     * @return array Array of allowed keys.
+     */
+    protected function getAllowedKeys()
+    {
+        return [
+            'xpath',
+            Tag::HTML,
+            Tag::HEAD,
+            Tag::BODY,
+            Attribute::CHARSET,
+            Attribute::VIEWPORT,
+            'ampElements',
+            'ampCustomStyle',
+            'ampCustomStyleByteCount',
+            'inlineStyleByteCount',
+            'links',
+            // Sometimes accessed by \Optimizer\Transformer\MinifyHtml::minifyNode().
+            // TODO: Revisit this.
+            'tagName',
```

Comment:
Interesting. This probably indicates a bug in `minifyNode`. Namely, it should be checking if the node is an instance of `Element` before attempting to access `tagName`.